service/controller/appcatalog/v1/resource/index/current_test.go

```diff
@@ -0,0 +1,125 @@
+package index
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/micrologger/microloggertest"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_Resource_GetCurrentState(t *testing.T) {
+	tests := []struct {
+		name              string
+		obj               *v1alpha1.AppCatalog
```

Comment:
It would be good to have `obj` as `interface{}` so we can also test for cases like `obj` is not an AppCatalog.

Reply:
It would be nice to add a test case where obj is an invalid value.